### PR TITLE
fix(pf): use close price for matching cryptowatch close timestamps

### PR DESCRIPTION
**Pull request by Reinis-FRP**
Signed-off-by: Reinis Martinsons <reinis@umaproject.org>

**Motivation**

Normally price feed should pick the last available price at or before request timestamp.

**Summary**

When price request timestamp matches close timestamp for Cryptowatch it now resolves to matching close price (previously it returned open price for the same period).

**Details**

Added check on whether price request timestamp matches end of OHLC period.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [X]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts [hunk 1]
```diff
@@ -189,7 +189,9 @@ export class CryptoWatchPriceFeed extends PriceFeedInterface {
     // why we prefer `before` periods to `after` ones. Similar reasoning is why we default to a period's open price to
     // its close price. We prefer prices that occurred before the target time.
     else {
-      returnPrice = this.invertPrice ? this._invertPriceSafely(match.openPrice) : match.openPrice;
+      // Use period close price when request time matches close time. In all other cases use open price instead.
+      const matchedPrice = time == match.closeTime ? match.closePrice : match.openPrice;
```
Comment (Reinis-FRP) [inline packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts:193]: I acknowledge not using strict equality operator as `time` is number, but `closeTime` is string. Should I have converted `closeTime` to number instead?

Comment (nicholaspai) [reply packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts:193]: I would convert them to the same type and use strict equality for completeness

Comment (nicholaspai) [reply packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts:193]: or just do time.toString()

**Review by nicholaspai — APPROVED** [review]
+1 LGTM

File: packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts (hunk 1 above)
Comment (Reinis-FRP) [reply packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts]: My assumptions were wrong and this was a problem only when running `getHistoricalPrice.js` and passing `--time` on comandline that. Next commit fixes that.

**Pull request merged**